MyOnlineStore/common-domain
# chore: Changed Doctrine Annotation to Attribute

## Summary 🌟
This changes the deprecated annotation docblocks to attribute. Used rector to change the blocks and then rearranged the comments there were needed manually.

## Technical choices ⚙️

- Introduced `3.x-dev` branch due to Doctrine `3.x` [not being BC](https://github.com/doctrine/orm/blob/3.0.x/UPGRADE.md).
- Used rector's `DoctrineSetList::ANNOTATIONS_TO_ATTRIBUTES` as-per [upgrade guide](https://www.doctrine-project.org/2022/11/04/annotations-to-attributes.html).
- Fixed a phpunit test regarding the currentcy code of Sierra Leone. The unit test relies on a list as provided by `moneyphp/money`. <sup>[[1]](https://en.wikipedia.org/wiki/Category:Currencies_with_ISO_4217_code)</sup>
   - This _was_ SLL and currently _is_ SLE <sup>[[2]](https://en.wikipedia.org/wiki/Sierra_Leonean_leone)</sup> 
   - Added the Bolívar Soberano (VED) from `moneyphp/money`
- Fixed the PHPUnit configuration deprecation: `1) Your XML configuration validates against a deprecated schema. Migrate your XML configuration using "--migrate-configuration"!`

## Testing instructions 📚
You can test this PR by following these steps:

1. You can use `myparcel-shipment-logger` to test this PR as it's currently using the attribute driver through this branch on [this PR](https://github.com/MyOnlineStore/myparcel-shipment-logger/pull/293). Proceed on that project with these testing instructions.
2. Make sure to clear the cache.
3. Make sure you've ran `make composer install`.
4. Run `make analyse` and observe that the StoreId class does not come up.
5. Run an `Myparcel` -> `ShipmentLogger` -> `Shipments` insomnia call (that uses the StoreId).

---

## Manual running test CI
The reason we run these manually is exceptional. This repo does not have access to the runners. I will run these all by hand and post the results here.

**Note:** I will get this fixed on a separate track with csi.

<details>
  <summary>PHPUnit</summary>

   ```
04ff9400b217:/srv/app/vendor/myonlinestore/common-domain$ ./vendor/bin/phpunit --migrate-configuration
PHPUnit 10.5.10 by Sebastian Bergmann and contributors.

Created backup:         /srv/app/vendor/myonlinestore/common-domain/phpunit.xml.dist.bak
Migrated configuration: /srv/app/vendor/myonlinestore/common-domain/phpunit.xml.dist
04ff9400b217:/srv/app/vendor/myonlinestore/common-domain$ ./vendor/bin/phpunit
PHPUnit 10.5.10 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.8
Configuration: /srv/app/vendor/myonlinestore/common-domain/phpunit.xml.dist

...............................................................  63 / 355 ( 17%)
............................................................... 126 / 355 ( 35%)
............................................................... 189 / 355 ( 53%)
............................................................... 252 / 355 ( 70%)
............................................................... 315 / 355 ( 88%)
........................................                        355 / 355 (100%)

Time: 00:00.107, Memory: 16.00 MB

OK (355 tests, 959 assertions)
   ```
</details>

<details>
  <summary>Psalm</summary>

   ```
[rob@SoaringDragon configuration]$ make attach php
04ff9400b217:/srv/app# ./vendor/bin/psalm
Target PHP version: 8.2 (inferred from current PHP version).
Scanning files...
Analyzing files...

░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░  60 / 165 (36%)
░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ 120 / 165 (72%)
░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░

------------------------------
                              
       No errors found!       
                              
------------------------------
60 other issues found.
You can display them with --show-info=true
------------------------------

Checks took 7.81 seconds and used 259.238MB of memory
Psalm was able to infer types for 97.3822% of the codebase
   ```
</details>
